src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -149,10 +149,9 @@ public class UserAccountManager implements IUserAccountManager {
   private final SchoolListReader schoolListReader;
 
   private final Pattern restrictedSignupEmailRegex;
-  private static final int USER_NAME_MAX_LENGTH = 255;
-  private static final Pattern USER_NAME_PERMITTED_CHARS_REGEX =
-      Pattern.compile("^[\\p{L}\\d_\\-' ]+$", Pattern.CANON_EQ);
+  private static final int USER_NAME_MAX_LENGTH = 50;
   private static final Pattern EMAIL_PERMITTED_CHARS_REGEX = Pattern.compile("^[a-zA-Z0-9!#$%&'+\\-=?^_`.{|}~@]+$");
+  private static final Pattern USER_NAME_PERMITTED_CHARS_REGEX = Pattern.compile("^[\\p{L}\\-'`! ]+$");
```

Comment:
This new regex now allows numbers to be used in user names (first / last name) - why? Backticks and exclamation marks are also now permitted, where they weren't before; they shouldn't be needed in a name?

It has also removed the ability to use an underscore (though I'm assuming that one is intentional?)

Reply:
Back ticks are invalid now, ! & _ are allowed